src/middlewared/middlewared/service.py

```diff
@@ -556,6 +562,104 @@ async def check_dependencies(self, id, ignored=None):
                             {'dependencies': list(dependencies.values())})
 
 
+class SharingTaskService(CRUDService):
+
+    path_field = 'path'
+    enabled_field = 'enabled'
+    locked_field = 'locked'
+    service_type = NotImplemented
+    locked_alert_class = NotImplemented
+    share_task_type = NotImplemented
+
+    @private
+    async def sharing_task_extend_context(self, extra):
+        return {
+            'locked_datasets': await self.middleware.call('zfs.dataset.locked_datasets'),
+            'service_extend': (await self.middleware.call(self._config.datastore_extend_context, extra))
+            if self._config.datastore_extend_context else {}
+        }
+
+    @private
+    async def validate_path_field(self, data, schema, verrors=None):
+        verrors = verrors or ValidationErrors()
```

Comment:
Which implementation uses this?

Reply:
Some child classes were not validating the path field at all, so I updated them to use this. This can however easily be done in each child class as well but seemed nicer to do it like this. If you have concerns, we can maybe do this in each child class ?

Reply:
No, that's good thing that you've added this validation, I've meant that I don't see any caller not passing `verrors` hence I don't see why do we need `verrors=None` default